cmake/modules/AddPureSwift.cmake

```diff
@@ -172,13 +172,15 @@ function(add_pure_swift_host_library name)
 
   # Make sure we can use the host libraries.
   target_include_directories(${name} PUBLIC
-    ${SWIFT_HOST_LIBRARIES_DEST_DIR})
+    "${SWIFT_HOST_LIBRARIES_DEST_DIR}")
+  target_link_directories(${name} PUBLIC
+    "${SWIFT_HOST_LIBRARIES_DEST_DIR}")
```

Comment:
This seems to be some difference between `lld` and `ld64`. `ld64` seems perfectly happy if the shared library is specified on the command line and there's a `-l<thatLib>` in `LC_LINKER_OPTION` but `lld` certainly isn't. It's the same auto linking issue as https://github.com/apple/swift/pull/64409. We didn't hit this before because we were adding the `lib/swift/host` *from the early swift syntax build directory* (and also only people using `lld` would hit it anyway).